bin/setup

```diff
@@ -1,72 +1,81 @@
-    system!('sudo apt install libpq-dev')
-  else
-    abort "\nUnable to detect OS {install_pg_dev}\n"
-  end
-end
-
-def can_install_postgres?
-  puts "Checking if environment can install 'pg' gem..."
-  # This evaluates to false if it fails
-  system('gem install pg')
-end
 
-def ruby_version_ok?
-  expected_version = `cat .ruby-version`.chomp
-  current_version = `ruby -v`.chomp
-  current_version.include?(expected_version)
-end
+require "fileutils"
 
 # path to your application root.
-APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
+APP_ROOT = File.expand_path("..", __dir__)
 
 def system!(*args)
   system(*args) || abort("\n== Command #{args} failed ==")
 end
 
-chdir APP_ROOT do
-  # This script is a starting point to setup your application.
-  # Add necessary setup steps to this file.
+#
+# Prints the log message with a yellow color to distinguish logs
+# See more information here - https://stackoverflow.com/questions/1489183/how-can-i-use-ruby-to-colorize-the-text-output-to-a-terminal
+#
+def log(message, color: :yellow)
+  color_code = if color == :yellow
+                 "33"
+               elsif color == :red
+                 "31"
+               else
+                 "37"
+               end
+
+  printf "\033[1;#{color_code}m";
+
+  puts "\n[ bin/setup ] #{message}"
+
+  printf "\033[0m"
+end
+
+
+FileUtils.chdir APP_ROOT do
+  log "---------------------"
+  log "❤️  Setup started... ❤️"
+  log "---------------------"
+
+  log "Encountered any issues with setup?\n If so, please open an issue @ https://github.com/rubyforgood/human-essentials/issues/new/choose", color: :red
+
+  #
+  # Validate that the ruby version requirement is met.
+  #
+  expected_version = `cat .ruby-version`.chomp
+  current_version = `ruby -v`.chomp
+  unless current_version.include?(expected_version)
+    log "Ruby version must be #{expected_version}. You are on #{current_version}"
+    exit
+  end
 
-  puts '== Installing dependencies =='
-  system! 'gem install bundler --conservative'
-  install_pg_dev unless can_install_postgres?
-  system('bundle check') || system!('bundle install')
+  log "== Installing dependencies =="
+  system! "gem install foreman"
+  system! "gem install bundler --conservative"
+  system!("bundle check") || system!("bundle install")
 
   # Install JavaScript dependencies if using Yarn
-  puts 'Setting up yarn'
+  log '== Installing javascript dependencies via yarn == '
   system('bin/yarn')
 
-  # puts "\n== Copying sample files =="
-  # unless File.exist?('config/database.yml')
-  #   cp 'config/database.yml.sample', 'config/database.yml'
-  # end
+  unless File.exist?('.env')
+    log "== Setup .env file from .env.example =="
+    system!('cp .env.example .env')
+  end
 
-  puts "\n== Preparing database =="
+  log "== Preparing database =="
+  log "⚠️  If you use docker to run postgres, make sure your database is running ⚠️"
+  system! 'bin/rails db:environment:set RAILS_ENV=development'
```

Comment:
I'm getting the following error at this step:
```ruby
rails aborted!
ActiveRecord::NoDatabaseError: We could not find your database: diaper_dev. Which can be found in the database configuration file located at config/database.yml.

To resolve this issue:

- Did you create the database for this app, or delete it? You may need to create your database.
- Has the database name changed? Check your database.yml config has the correct database name.

To create your database, run:

        bin/rails db:create


Caused by:
PG::ConnectionBad: connection to server at "::1", port 5432 failed: FATAL:  database "diaper_dev" does not exist

Tasks: TOP => db:environment:set
(See full trace by running task with --trace)
```

Looks like the database needs to exist in order to be able to run ` db:environment:set` (IDK why)
I was able to reproduce the error by dropping the database via `bin/rails db:drop` and then `bin/rails db:environment:set RAILS_ENV=development`

To be clear, a database with the name specified in `config/database.yml` needs to exist for `bin/rails db:environment:set RAILS_ENV=development` to run successfully. Someone who just cloned the repo won't have that database created yet by the time the environment set command is executed since the `bin/rails db:setup` command is in charge of creating the database with the required name and it is executed after the environment set command.

Reply:
Gotcha. So I think we need to create the DB first? but wouldn't dropping make sense first so you know you are operating on a clean DB?

Reply:
@heyapricot -- I realized the problem that this line of code tries to address is more specific to me or anyone who has cloned production DB into local for debugging purposes. That being said, I've removed this line of code and would appreciate you running the script again :).

I'am hoping to finally merge this in soon.